vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -320,215 +321,303 @@ func (e *ProposalTxExecutor) RewardValidatorTx(tx *txs.RewardValidatorTx) error
+	// the validator they are delegated to.
+	validator, err := e.OnCommitState.GetCurrentValidator(delegator.SubnetID, delegator.NodeID)
+	if err != nil {
+		return fmt.Errorf("failed to get whether %s is a validator: %w", delegator.NodeID, err)
+	}
+
+	vdrTxIntf, _, err := e.OnCommitState.GetTx(validator.TxID)
+	if err != nil {
+		return fmt.Errorf("failed to get whether %s is a validator: %w", delegator.NodeID, err)
+	}
+
+	// Invariant: Delegators must only be able to reference validator
+	//            transactions that implement [txs.ValidatorTx]. All
+	//            validator transactions implement this interface except the
+	//            AddSubnetValidatorTx.
+	vdrTx, ok := vdrTxIntf.Unsigned.(txs.ValidatorTx)
+	if !ok {
+		return ErrWrongTxType
+	}
+
+	// Calculate split of reward between delegator/delegatee
+	delegatorReward, delegateeReward := reward.Split(delegator.PotentialReward, vdrTx.Shares())
 
-		vdrTxIntf, _, err := e.OnCommitState.GetTx(vdrStaker.TxID)
+	utxosOffset := 0
+
+	// Reward the delegator here
+	reward := delegatorReward
+	if reward > 0 {
+		rewardsOwner := uDelegatorTx.RewardsOwner()
+		outIntf, err := e.Fx.CreateOutput(reward, rewardsOwner)
 		if err != nil {
-			return fmt.Errorf(
-				"failed to get whether %s is a validator: %w",
-				stakerToRemove.NodeID,
-				err,
-			)
+			return fmt.Errorf("failed to create output: %w", err)
 		}
-
-		// Invariant: Delegators must only be able to reference validator
-		//            transactions that implement [txs.ValidatorTx]. All
-		//            validator transactions implement this interface except the
-		//            AddSubnetValidatorTx.
-		vdrTx, ok := vdrTxIntf.Unsigned.(txs.ValidatorTx)
+		out, ok := outIntf.(verify.State)
 		if !ok {
-			return ErrWrongTxType
+			return ErrInvalidState
 		}
-
-		// Calculate split of reward between delegator/delegatee
-		// The delegator gives stake to the validatee
-		validatorShares := vdrTx.Shares()
-		delegatorShares := reward.PercentDenominator - uint64(validatorShares)                            // parentTx.Shares <= reward.PercentDenominator so no underflow
-		delegatorReward := delegatorShares * (stakerToRemove.PotentialReward / reward.PercentDenominator) // delegatorShares <= reward.PercentDenominator so no overflow
-		// Delay rounding as long as possible for small numbers
-		if optimisticReward, err := math.Mul64(delegatorShares, stakerToRemove.PotentialReward); err == nil {
-			delegatorReward = optimisticReward / reward.PercentDenominator
+		utxo := &avax.UTXO{
+			UTXOID: avax.UTXOID{
+				TxID:        txID,
+				OutputIndex: uint32(len(outputs) + len(stake)),
+			},
+			Asset: stakeAsset,
+			Out:   out,
 		}
-		delegateeReward := stakerToRemove.PotentialReward - delegatorReward // delegatorReward <= reward so no underflow
 
-		offset := 0
+		e.OnCommitState.AddUTXO(utxo)
+		e.OnCommitState.AddRewardUTXO(txID, utxo)
+
+		utxosOffset++
+	}
+
+	// Reward the delegatee here
+	if delegateeReward > 0 {
```

Comment:
Can we inverse this check and reduce the indentation?

Reply:
Done